tooling/compare-builds.sh

```diff
@@ -0,0 +1,604 @@
+    prepareAlternateJavaHome "${JDK_ID}" "${LAST_DOWNLOADED_FILE}"
+  fi
+done
+
+if uname | grep CYGWIN ; then
+  initWindowsTools
+  setMsvscToPath
+  compileAndSetToFuturePath
+fi
+
+# comapre build can not run if not run from its pwd
+pushd "${COMPARE_WAPPER_SCRIPT_DIR}/reproducible/"
+  if [ "${COMP_TYPE}" = "COMP" ] ; then
+    for jdkName in ${JDK_INFO["first_name"]} ${JDK_INFO["second_name"]} ; do
+      # better to try them asap
+      "${WORKDIR}/${jdkName}/bin/java" --version 
+      "${WORKDIR}/${jdkName}/bin/javac" --version
+      ftureDir="$(mktemp -d)/classes"
+      if uname | grep CYGWIN ; then
+        ftureDir=$(cygpath -m "${ftureDir}")
+      fi
+      # the java in java_home will be used later, try it
+      "$JAVA_HOME/bin/java" --version
+      "$JAVA_HOME/bin/javac" -d "${ftureDir}" "../../tooling/src/java/temurin/tools/BinRepl.java"
+      CLASSPATH="${ftureDir}"
+      export CLASSPATH
+      if uname | grep CYGWIN ; then
+        CLASSPATH=$(cygpath -m "${CLASSPATH}")
+        export CLASSPATH="${CLASSPATH}"
+      fi
+      getKeysForComparablePatch "${WORKDIR}/${jdkName}"
+      # we cannot use the JDK we are currently processing. It is broken by the patching
+      PATH="${JAVA_HOME}/bin:${PATH}" bash ./comparable_patch.sh --jdk-dir "${WORKDIR}/${jdkName}" --version-string "$JAVA_VENDOR_VERSION" --vendor-name "$JAVA_VENDOR" --vendor_url "$JAVA_VENDOR_URL" --vendor-bug-url "$JAVA_VENDOR_URL_BUG" --vendor-vm-bug-url "$JAVA_VENDOR_URL_BUG"  2>&1 | tee -a "$LOG"
+    done
+  fi
+  if [ "${DO_BREAK:-}" == "true" ] ; then
+    echo "dead" > "${WORKDIR}/${JDK_INFO["first_name"]}/bin/java"
+    echo "dead" > "${WORKDIR}/${JDK_INFO["first_name"]}/lib/server/libjvm.so"
+  fi
+  GLOABL_RESULT=0
+  if [ "${COMP_TYPE}" = "COMP" ] ; then
+    # this tells repro_compare to skipp all the preprocessing. If not set, the repro_compare.sh on top of comparable_patch.sh  brings false negatives
+    export PREPROCESS="no"
+  fi
+  bash ./repro_compare.sh openjdk "${WORKDIR}/${JDK_INFO["first_name"]}" openjdk "${WORKDIR}/${JDK_INFO["second_name"]}" 2>&1 | tee -a "${LOG}" || GLOABL_RESULT=$?
+  diflog="${WORKDIR}/differences.log"
+  totlog="${WORKDIR}/totalfiles.log"
+  set +e
+    {
+      grep "Number of differences" -A 1 "${LOG}"
+      echo "Warning, the files which are only in one of the JDKs may be missing in this summary:"
+      grep "Only in" "${LOG}" | sed "s|${WORKDIR}||g"
+      echo -n "Only in: "
+      ## shellcheck disable=SC2126
+      grep -c "Only in " "${LOG}"
+    } > "${diflog}"
+    grep "Number of files" "${LOG}" > "${totlog}"
+  set -e
+popd
+
+diffFile="${COMPARE_WAPPER_SCRIPT_DIR}/reproducible/reprotest.diff"
+tapsAndJunits "${diffFile}" "${diflog}" "${totlog}"
+cp "${diffFile}" "${WORKDIR}"
+pwd
+ls -l "$(pwd)"
+
+if [ ! "${DO_RESULTS:-}" == "false" ] ; then
+  # the result will be changed via tap plugin or jtreg plugin
+  exit 0
+else
+  exit ${GLOABL_RESULT}
```

Comment:
Probably ought to spell this?: GLOBAL_RESULT

Reply:
Ouch. sure!